File: src/ast/ConversionExpression.java
package ast;

import java.io.StringWriter;

public class ConversionExpression extends BaseNode {
    private BaseNode _typeNode;
    private BaseNode _expressions;

    public ConversionExpression(BaseNode typeNode, BaseNode expressions) {
        super(NodeType.ConversionExpression);
        _typeNode = typeNode;
        _expressions = expressions;
    }

    @Override
    public void PrintLeft(StringWriter writer) {
        writer.write("(");
        _typeNode.Print(writer);
        writer.write(")(");
        _expressions.Print(writer);
    }
}

File: src/de/fearlesstobi/demangler/ast/ConversionExpression.kt
package de.fearlesstobi.demangler.ast

import java.io.StringWriter

class ConversionExpression(private val typeNode: BaseNode, private val expressions: BaseNode) : BaseNode(NodeType.ConversionExpression) {
    public override fun printLeft(writer: StringWriter) {
        writer.write("(")
        typeNode.print(writer)
        writer.write(")(")
        expressions.print(writer)
    }

}